### PR TITLE
[Minor] Fix stage partitioning algorithm

### PR DESCRIPTION
**Pull request by wonook**
This PR:

- Fixes the stage partitioning algorithm
  - gets rid of the part that considers previous dependent stages and replaces it with stage boundaries with multiple inputs
- A bit of refactoring in ScheduleGroupPass
- Fix test to meet the new changes
- Fix log messages to reduce the amount of verboseness.

### CODE REVIEW

File: runtime/common/src/main/java/edu/snu/onyx/runtime/common/message/ncs/NcsMessageSender.java [hunk 1]
```diff
@@ -27,13 +27,13 @@
 
   @Override
   public void send(final ControlMessage.Message message) {
-    LOG.debug("send: {}", message);
+    LOG.trace("send: {}", message);
```
Comment (sanha) [inline runtime/common/src/main/java/edu/snu/onyx/runtime/common/message/ncs/NcsMessageSender.java]: If you'd like to change this, please change the logging way of `NcsMessageEnvironment`, `GrpcMessageClient`, `GrpcMessageContext`,... and so on also.

**Comment by wonook** [comment]
!rebuild

File: runtime/common/src/main/java/edu/snu/onyx/runtime/common/message/ncs/NcsMessageSender.java (hunk 1 above)
Comment (wonook) [reply runtime/common/src/main/java/edu/snu/onyx/runtime/common/message/ncs/NcsMessageSender.java]: Sure.

**Comment by wonook** [comment]
!rebuild

**Comment by wonook** [comment]
!rebuild

**Comment by wonook** [comment]
!rebuild

**Comment by wonook** [comment]
!rebuild

**Comment by wonook** [comment]
!rebuild

**Pull request merged**